src/us/kbase/workspace/gitcommit/GitCommit.java

```diff
@@ -0,0 +1,38 @@
+package us.kbase.workspace.gitcommit;
+
+import java.io.InputStream;
+import java.util.Scanner;
+
+/** The Git commit from which the service was built. Expects a file called "gitcommit" in the same
+ * directory as the class file which contains the commit hash.
+ * 
+ * If the file is missing, the Git commit will be replaced with an error message.
+ * @author gaprice@lbl.gov
+ *
+ */
+public class GitCommit {
+	
+	// can't really test this easily since the file must be baked into the jar or war,
+	// just test manually
+
+	/** The Git commit from which the service was built. */
+	public static final String COMMIT;
+	
+	private static final String COMMIT_FILE_NAME = "gitcommit";
+	
+	static {
```

Comment:
how is static {} different from `static` keyword before var or method

Reply:
A static block runs when the class is imported for the first time. A static variable or method are class variables / methods

Reply:
https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html

Reply:
Make sense now!!